src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -73,19 +93,33 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
             update_ids_method(one_proc_info);
             stream_info[NUMBER_OF_STREAMS] =
                 static_cast<int>(one_proc_info[proc_type] / stream_info[THREADS_PER_STREAM]);
-            if ((stream_info[NUMBER_OF_STREAMS] == 0) && (proc_type == MAIN_CORE_PROC)) {
-                stream_info[NUMBER_OF_STREAMS] =
-                    static_cast<int>((one_proc_info[MAIN_CORE_PROC] + one_proc_info[HYPER_THREADING_PROC]) /
-                                     stream_info[THREADS_PER_STREAM]);
-            }
             if (n_streams < stream_info[NUMBER_OF_STREAMS]) {
                 stream_info[NUMBER_OF_STREAMS] = n_streams;
             }
-            streams_info_table.push_back(stream_info);
+            if (stream_info[NUMBER_OF_STREAMS] > 0) {
+                streams_info_table.push_back(stream_info);
+                n_streams -= stream_info[NUMBER_OF_STREAMS];
+            }
+        }
+    };
 
-            n_streams -= stream_info[NUMBER_OF_STREAMS];
-            proc_socket_table[one_proc_info[PROC_SOCKET_ID]][proc_type] -=
-                stream_info[NUMBER_OF_STREAMS] * stream_info[THREADS_PER_STREAM];
+    auto check_threads_per_stream = [&]() {
+        int count = 0;
+        while (count < n_streams) {
+            for (int n_type = MAIN_CORE_PROC; n_type <= HYPER_THREADING_PROC; n_type++) {
+                count += static_cast<int>(proc_type_table[0][n_type] / n_threads_per_stream);
+            }
+            if (((count == 0) && (proc_type_table.size() > 1)) || (count >= n_streams)) {
+                return;
```

Comment:
It seem `count >= n_streams` is duplicated with `while (count < n_streams)`

Reply:
`((count == 0) && (proc_type_table.size() > 1)` should be abnormal condition, do we need handle it rather than return directly?

Reply:
updated